MIGRATION.md

```diff
@@ -0,0 +1,130 @@
+
+## Helper
+
+This release includes v3 of the `algoliasearch-helper` package. If you are using the built-in widgets, nothing will change for you. This version no longer includes `lodash`, having two main places of impact for users. The main difference will be in bundle size, which is now significantly smaller. If you are using any methods from the `helper` or `searchResults`, please see the detailed changelog [here](https://github.com/algolia/algoliasearch-helper-js/blob/next/documentation-src/metalsmith/content/upgrade.md).
+
+## Custom widgets
+
+As mentioned earlier, any reference of the helper now contains the "new helper", as well as the `getConfiguration` life cycle no longer being used. You can replace its usage with `getWidgetSearchParameters` and `getWidgetState`. Note that this means that your custom widget will also take part in routing from then on. You can still exclude it from the URL via `stateMapping`.
+
+### getConfiguration
+
+This life cycle could be used for setting up a facet, or a default refinement previously. This is now replaced by `getWidgetState` and `getWidgetSearchParameters`. This means that there's no longer a separate step for setting up. However, that can be implemented using a check inside `getWidgetSearchParameters`. Code like this (abbreviated):
+
+```js
+const widget = {
+  getConfiguration(searchParams) {
+    return {
+      disjunctiveFacets: ['myAttribute'],
+    };
+  },
+};
+```
+
+Will become like this:
+
+```js
+const widget = {
+  getWidgetSearchParameters(searchParameters, { uiState }) {
+    return searchParameters
+      .addDisjunctiveFacet('myAttribute')
+      .addDisjunctiveFacetRefinement(
+        'myAttribute',
+        uiState.myWidgetName.myAttribute
+      );
+  },
+  getWidgetState(uiState, { searchParameters }) {
+    return {
+      ...uiState,
+      myWidgetName: {
+        myAttribute: searchParameters.getDisjunctiveRefinements('myAttribute')
+      }
+    };
+  }
+};
+```
+
+While this looks longer, more of the functions are now together than before, and it isn't possible anymore to get into a state where `getConfiguration` is called multiple times and isn't behaving as you would want.
+
+## connectAutoComplete
+
+The `indices` option has been removed, in favour of using `index` widgets (see the federated search section). In practice, this:
+
+```js
+const autocomplete = connectAutocomplete(() => {/* ... */});
+
+search.addWidget(
+  autocomplete({
+    indices: [{
+      name: "additional"
+    }]
+  })
+);
+```
+
+Will be replaced with this:
+
+```js
+const autocomplete = connectAutocomplete(() => {/* ... */});
+
+search.addWidgets([
+  index({ indexName: "additional" }),
+  autocomplete()
+]);
+```
```

Comment:
as discussed offline, I would love to have an example where this cannot be entirely replaced with a simple searchBox. I'm still missing it.

Reply:
What about attaching a link to the source code of the storybook and its live example?
I just tried storybook and I think it explain well how it works.

Reply:
you have access to the hits of both of the indices in autocomplete, so it's not the same as a search box (if the autocomplete component you're wrapping is using both hits & query at the same time, which is common)